planner/core/rule_join_elimination.go

```diff
@@ -173,38 +163,71 @@ func (o *outerJoinEliminator) isDuplicateAgnosticAgg(p LogicalPlan) (_ bool, col
 			return false, nil
 		}
 		for _, expr := range aggDesc.Args {
-			if col, ok := expr.(*expression.Column); ok {
-				cols = append(cols, col)
-			} else {
-				return false, nil
-			}
+			cols = append(cols, expression.ExtractColumns(expr)...)
 		}
 	}
 	return true, cols
 }
 
-func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentSchema *expression.Schema) (LogicalPlan, error) {
+func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentCols []*expression.Column) (LogicalPlan, error) {
+	var err error
+	var isEliminated bool
+	for join, isJoin := p.(*LogicalJoin); isJoin; join, isJoin = p.(*LogicalJoin) {
+		p, isEliminated, err = o.tryToEliminateOuterJoin(join, aggCols, parentCols)
+		if err != nil {
+			return p, err
+		}
+		if !isEliminated {
+			break
+		}
+	}
+
+	var colsInSchema []*expression.Column
+	switch x := p.(type) {
+	case *LogicalProjection:
+		for _, expr := range x.Exprs {
+			colsInSchema = append(colsInSchema, expression.ExtractColumns(expr)...)
+		}
+		for i := len(aggCols) - 1; i >= 0; i-- {
+			idx := p.Schema().ColumnIndex(aggCols[i])
+			if idx == -1 {
+				aggCols = append(aggCols[:i], aggCols[i+1:]...)
+				continue
+			}
+			if col, ok := x.Exprs[idx].(*expression.Column); ok {
+				aggCols[i] = col
+			} else {
+				aggCols = append(aggCols[:i], aggCols[i+1:]...)
```

Comment:
Should we replace the `aggCols[i]` with the columns extracted from `p.Exprs[idx]`?

Reply:
Actually, we don't need to modify the `aggCols` now. Because `isParentColsAllFromOuter` has been checked for the existing aggregation situation.